apps/ledger-live-mobile/src/families/evm/EditTransactionFlow/MethodSelection.tsx

```diff
@@ -0,0 +1,240 @@
+import { ScreenName } from "../../../const";
+import { EditTransactionParamList } from "./EditTransactionParamList";
+
+const getSpeedUpDescriptionKey = (
+  haveFundToSpeedup: boolean,
+  isOldestEditableOperation: boolean,
+):
+  | "editTransaction.speedUp.description"
+  | "editTransaction.error.notEnoughFundsToSpeedup"
+  | "editTransaction.error.notlowestNonceToSpeedup" => {
+  if (!haveFundToSpeedup) {
+    return "editTransaction.error.notEnoughFundsToSpeedup";
+  }
+
+  if (!isOldestEditableOperation) {
+    return "editTransaction.error.notlowestNonceToSpeedup";
+  }
+
+  return "editTransaction.speedUp.description";
+};
+
+type Props = StackNavigatorProps<
+  EditTransactionParamList,
+  ScreenName.EditTransactionMethodSelection
+>;
+
+function MethodSelectionComponent({ navigation, route }: Props) {
+  const { t } = useTranslation();
+  const { operation, account, parentAccount } = route.params;
+
+  invariant(
+    operation.transactionRaw,
+    "operation.transactionRaw not found. Could not edit the transaction",
+  );
+
+  const [transactionHasBeenValidated, setTransactionHasBeenValidated] = useState(false);
+
+  const mainAccount = getMainAccount(account, parentAccount);
+
+  const transactionToEdit = fromTransactionRaw(
+    operation.transactionRaw as TransactionRaw,
+  ) as EvmTransaction;
+
+  const { transaction, setTransaction } = useBridgeTransaction<EvmTransaction>(() => ({
+    account,
+    parentAccount,
+    transaction: transactionToEdit,
+  }));
+
+  invariant(
+    transaction,
+    "[useBridgeTransaction - MethodSelection] could not found transaction from bridge.",
+  );
+
+  const haveFundToCancel = hasMinimumFundsToCancel({
+    mainAccount,
+    transactionToUpdate: transactionToEdit,
+  });
+
+  const haveFundToSpeedup = hasMinimumFundsToSpeedUp({
+    account,
+    mainAccount,
+    transactionToUpdate: transactionToEdit,
+  });
+
+  const [selectedMethod, setSelectedMethod] = useState<EditType | null>();
+
+  // if we are at this step (i.e: in this screen) it means the transaction is editable
+  const isOldestEditableOperation = isOldestPendingOperation(mainAccount, transactionToEdit.nonce);
+  const bridge: AccountBridge<EvmTransaction> = getAccountBridge(account, parentAccount as Account);
+
+  const onSelect = useCallback(
+    async (option: EditType) => {
+      log("Edit Transaction - Method Selection", "onSelect Cancel/Speed up", option);
+
+      const patch = await getEditTransactionPatch({
+        account: mainAccount,
+        transaction: transactionToEdit,
+        editType: option,
+      });
+
+      setTransaction(bridge.updateTransaction(transaction, patch));
+
+      setSelectedMethod(option);
+    },
+    [mainAccount, transaction, transactionToEdit, bridge, setTransaction],
+  );
+
+  /**
+   * Poll for transaction confirmation
+   * If the transaction has been confirmed, we navigate to the error screen
+   * PS: this still runs even if the user has navigated forward in the steps
```

Comment:
When does it stop then ? It doesn't run after you leave the flow I guess ? So after broadcast ?
Maybe polling every n seconds (30 seconds by default) is a bit overkill as users will rarely stay on the step for minutes, if you ask once at every step if the transaction is confirmed or not it should be enough to catch.

Reply:
The polling stops when:
- the tx has been confirmed on the blockchain (`tx?.confirmations` is truthy)
- the component unmount (cf. `return` in `useEffect`), i.e the users exits the "EditTransactionFlow" stack

> Maybe polling every n seconds (30 seconds by default) is a bit overkill as users will rarely stay on the step for minutes, if you ask once at every step if the transaction is confirmed or not it should be enough to catch.

This was the original implementation but it's (IMHO) a naive and suboptimal implementation. Here is why:
Most EVM blockchains have pretty fast block times (i.e: time between each new block).
For ethereum for example, this is between 12- 15 seconds (probably more for other blockchains). A user is likely to stay more than 12-15 seconds on a step (mainly the fee step if he wants to update them, or the summary step if he is diligent in his verification).
In addition to that, verifying only once at each step, as originally done, would imply an (excessive) code duplication of this check logic at every step (with associated complexity in maintenance and potential issues / oversight if the flow is updated and a step is added for example).

The idea here is to centralize this verification process in one place, at the first step.
And since this flow is a stack navigation, this logic still runs when the user goes past this method selection step since what React Native do under the hood is adding a new screen (for the next step) on top of the previous once (hence the "stack" navigation name).

Reply:
Alright sounds good!